### DISCUSSION TITLE
Authenticating in Jupyter Notebook

### DISCUSSION THREAD

**Discussion by michaelmech** [areed1192/td-ameritrade-python-api#179] [Q&A] [answered]
Hi. I've pip installed td-ameritrade-python-api into a jupyter notebook and was trying to follow the instructions for authenticating from the pypi site. Unfortunately, the instructions seemed to have in mind the use of an entire python script and terminal. Is there a way to authenticate from jupyter notebook?

**Comment by eyesuk** [comment]
If you are using VS code you can use the Python interactive mode which essentially turns your code into a Jupiter notebook
You can do this in any line by typing `# %%`
![image](https://user-images.githubusercontent.com/9576885/116689906-a7bfaa80-a986-11eb-91bd-652e92b56be2.png)

**Comment by michaelmech** [comment]
I was able to do that, but when I try to run some simple functions of the library, it returns None. For example:

![github_image](https://user-images.githubusercontent.com/56457392/116734696-6e953380-a9a2-11eb-8bb5-0f83f4d0bae8.JPG)

Based on reading other "Issues" in this github, I think it has to do with the manual reauthentication(copying the redirect url from the search bar) but the instructions for it only work for a terminal...

**Comment by michaelmech** [comment, marked as answer]
Nevermind. I believe I solved it. I just looked at the code, and manually used the 'oath' function in the notebook. Then I followed the instructions of taking the url and putting it into the input() bar in order for the function to parse through it.
